### PR TITLE
Add CREATE DATABASE command to SQL

### PR DESCRIPTION
**Pull request by matthewmturner**
# Which issue does this PR close?

Second task from #1877 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Looking good from here

**Review by jychen7 — COMMENTED** [review]
LGTM

**Comment by matthewmturner** [comment]
Just need to add a test

**Comment by matthewmturner** [comment]
Im a little confused why getting a `ParserError` - it seems i will need to look into this more on the sqlparser-rs side.

**Comment by matthewmturner** [comment]
i think im going to need to add some logic for properly adding a schema to a catalog

**Comment by matthewmturner** [comment]
Given there isnt a sql parameter in postgres for specifying the database for a schema (it uses the current active db - https://www.postgresql.org/docs/current/sql-createschema.html) i think we could do one or both of the following:

- to align with postgres add the notion of active catalog and create schema within that.  this will require adding a command to update the active catalog.
- qualify the schema names.  i.e. `CREATE SCHEMA abc` continues in current / active catalog but then we also allow `CREATE SCHEMA my_catalog.abc` which would create a schema in the specified catalog.

i personally think the second approach would align well with how catalogs / schemas are set using `CREATE TABLE`.  Then we could always add more postgres like functionality as a second step.

@alamb @jychen7 what do you think?

**Comment by alamb** [comment]
> i personally think the second approach would align well with how catalogs / schemas are set using CREATE TABLE. Then we could always add more postgres like functionality as a second step.

I agree the second approach seems reasonable 👍

**Comment by jychen7** [comment]
> qualify the schema names. i.e. CREATE SCHEMA abc continues in current / active catalog but then we also allow CREATE SCHEMA my_catalog.abc which would create a schema in the specified catalog.

yeah, make sense

**Comment by matthewmturner** [comment]
@alamb I believe this is ready

File: datafusion/core/src/logical_plan/plan.rs [hunk 1]
```diff
@@ -363,6 +374,8 @@ pub enum LogicalPlan {
     CreateMemoryTable(CreateMemoryTable),
     /// Creates a new catalog schema.
     CreateCatalogSchema(CreateCatalogSchema),
+    /// Creates a new catalog schema.
```
Comment (alamb) [inline datafusion/core/src/logical_plan/plan.rs:377]:
```suggestion
    /// Creates a new catalog (aka "Database").
```

File: datafusion/core/src/logical_plan/plan.rs [hunk 2]
```diff
@@ -193,7 +193,18 @@ pub struct CreateExternalTable {
...
     /// The table schema
     pub schema_name: String,
-    /// The table name
+    /// Do nothing (except issuing a notice) if a schema with the same name already exists
+    pub if_not_exists: bool,
+    /// Empty schema
+    pub schema: DFSchemaRef,
+}
+
+/// Creates a catalog.
```
Comment (alamb) [inline datafusion/core/src/logical_plan/plan.rs:202]:
```suggestion
/// Creates a catalog (aka "Database")
```

File: datafusion/core/src/execution/context.rs [hunk 3]
```diff
@@ -3221,6 +3257,32 @@ mod tests {
...
+        // the information schema used to introduce cyclic Arcs
+        let ctx = SessionContext::with_config(
+            SessionConfig::new().with_information_schema(true),
+        );
+
+        // Create catalog
+        ctx.sql("CREATE DATABASE test").await?.collect().await?;
+
+        // Create schema
+        ctx.sql("CREATE SCHEMA test.abc").await?.collect().await?;
```
Comment (alamb) [inline datafusion/core/src/execution/context.rs:3271]: this is cool 👍

**Review by alamb — APPROVED** [review]
LGTM -- thanks @matthewmturner

**Review by yjshen — APPROVED** [review]

**Pull request merged**